internal/app/proxyd/main.go

```diff
@@ -48,46 +52,76 @@ func main() {
 	// Start up with initial bootstrap config
 	go r.Bootstrap(bootstrapCtx)
 
-	// nolint: errcheck
-	go func() {
-		kubeconfig := "/etc/kubernetes/admin.conf"
-		if err = conditions.WaitForFilesToExist(kubeconfig).Wait(context.Background()); err != nil {
-			log.Fatalf("failed to find %s: %v", kubeconfig, err)
-		}
+	go waitForKube(r)
 
-		config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
-		if err != nil {
-			log.Fatalf("failed to read config %s: %v", kubeconfig, err)
-		}
+	var wg sync.WaitGroup
 
-		// Discover local non loopback ips
-		ips, err := pkgnet.IPAddrs()
-		if err != nil {
-			log.Fatalf("failed to get local address: %v", err)
+	wg.Add(1)
+	// Start up reverse proxy
+	go func(r *frontend.ReverseProxy, wg *sync.WaitGroup) {
```

Comment:
We can simplify this with the ListenAndServe func in the factory. Checkout trustd.